### ISSUE TITLE
Update the injection code to make sure we add it outside of a comment block

### ISSUE THREAD

**Issue by Pomax** [mozilla/thimble.mozilla.org#1503]
While injection based on `</head>` generally works, the following code (reported in https://github.com/mozilla/thimble.mozilla.org/issues/1502) will cause incorrect script injection:
```
...
  ...
  <!--
  </head>
  -->
  </head>
  <body>
  ...
...
```
The current code will find the first `</head>` instance, and add the remix script, which thus ends up as
```
...
  ...
  <!--
  <script src="remix script"></script>
  </head>
  -->
  </head>
  <body>
  ...
...
```
In order to ensure the script is not injected inside a comment, we'll need to update the code so that comments get ignored for positioning purposes.

**Comment by khanh2003** [comment]
I had an Idea.
Just inject the remix script at the beginning of the html file.
for example:
```
  <script src="remix script"></script>
<meta charset="utf-8">
... blah blah blah
```

```
  <script src="remix script"></script>
--> <!-- the remix script is still working. -->
<meta charset="utf-8">
... blah blah blah
```
And remember, the user can always remove the remix button by

```
$(function() {document.body.innerHTML = "";}); // if JQuery is included.
```

**Comment by gideonthomas** [comment]
The ability to remove the remix button via hacks has been made intentionally available since sometimes, advanced projects can find the remix button a nuisance in the UI (for e.g. I made a teachable pacman game in which I had to remove the remix button through code otherwise pacman couldn't fully be seen). I don't think the idea here is to cover all these cases.

However, I do think that the comment block case should be fixed since it isn't much of a hack and the removal of the remix button in this case tends to be unintentional.

**Comment by humphd** [comment]
Doing this properly would require us to do more than text insertion, we would really need to construct the full DOM so we could properly detect comments.

**Comment by Pomax** [comment]
Not necessarily: writing a tokenizer that finds the first `</head>` that is not inside a comment block or script block isn't all that much work. But it does highlight another question: what do we do with html source that doesn't actually contain a `<head>` element (technically legal HTML5)?

**Comment by khanh2003** [comment]
Will we give up?
This is a hack, and every playground always has some ways to 
```
- comment the remix script (this bug)
- access <head> element (like jsfiddle, normally we can only access <body> element)
- remove "by _________"
```
These are needed.

**Comment by samueller** [comment]
The ability to easily remove the remix button should exist. The remix button is great, gives exposure to Thimble and encourages others to quickly learn and improve upon other projects. However, _a learner will very early on find it conflicting with their site design_, both in the aesthetics and functionality. We teach **kids** navigation bars soon into our curriculum. As you can see, the remix button is problematic:
https://thimbleprojects.org/ucode/87099/

**Comment by gideonthomas** [comment]
@flukeout can we get some design direction for the remix button toggle (probably in the publish dialog box)?
It should then be fairly trivial to implement the backend pieces for this once UI is done.

I'm going to move this discussion to a new issue since @Pomax's issue concerns a different UX problem.

**Comment by flukeout** [comment]
Sounds good @gideonthomas - I'll get on it. Agreed that we should make it optional. I'll post some UI wireframes here a bit later today and we can team up on implementation.

Will this be tracked as a field in the published projects database? It would be good to see how often this gets used.

One small concern is that once the button is removed, we give no indication of how the project was created. Is that important? Not sure, but worth giving some thought to other ways that we can indicate "Built with Thimble"

**Comment by gideonthomas** [comment]
@flukeout: https://github.com/mozilla/thimble.mozilla.org/issues/1515#issuecomment-233383843

**Comment by gideonthomas** [comment]
Going to define the scope of this issue to be specific to @Pomax's comment which suggests we need to make our injection code smarter and figure out cases when it might not be injected.
